src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1055,21 +1058,30 @@ impl EventProcessor {
         // Set the timestamp.
         wt.xconn.set_timestamp(event.time as xproto::Timestamp);
 
+        let Some(DeviceType::Mouse) = self
+            .devices
+            .borrow()
+            .get(&DeviceId(event.sourceid as xinput::DeviceId))
```

Comment:
This came as a surprise to me, but I had to do quite some digging to figure out why this wasn't working for me.
`deviceId` here uses the cursor, but `sourceId` actually uses the device that moved the cursor. Shouldn't we use that to send to the user?